Fix topology view when displaying mixed connect-native/normal services. (#13023)
hashicorp/consul: .changelog/13023.txt

```release-note:improvement

> ui: the topology view now properly checks for intentions with mixed connect native / normal services.

Reviewer: ```suggestion
ui: the topology view now properly displays services with mixed connect and non-connect instances.
```
